pkg/kubelet/kubelet_pods_test.go

```diff
@@ -2595,6 +2595,42 @@ func Test_generateAPIPodStatus(t *testing.T) {
 				},
 			},
 		},
+		{
```

Comment:
Add another condition that swaps pod and previousStatus, and then one that has restart never and all containers have currentStatus = terminated

Reply:
The first is already covered with https://github.com/kubernetes/kubernetes/blob/c32e56912852a2f902426394fdbe5950176e1eae/pkg/kubelet/kubelet_pods_test.go#L2563-L2597 but we don't have anything for restart never so adding that now

Reply:
Pushed one more unit test to cover the other case.